system/ui/lib/list_view.py

```diff
@@ -188,266 +296,56 @@ def _render(self, rect: rl.Rectangle) -> bool:
-  visible: bool | Callable[[], bool] = True
-
-  # Cached properties for performance
-  _prev_max_width: int = 0
-  _wrapped_description: str | None = None
-  _prev_description: str | None = None
-  _description_height: float = 0
-
-  @property
-  def is_visible(self) -> bool:
-    return bool(_resolve_value(self.visible, True))
-
-  def get_description(self):
-    return _resolve_value(self.description, None)
-
-  def get_item_height(self, font: rl.Font, max_width: int) -> float:
-    if not self.is_visible:
-      return 0
-
-    current_description = self.get_description()
-    if self.description_visible and current_description:
-      if (
-        not self._wrapped_description
-        or current_description != self._prev_description
-        or max_width != self._prev_max_width
-      ):
-        self._prev_max_width = max_width
-        self._prev_description = current_description
-
-        wrapped_lines = wrap_text(font, current_description, ITEM_DESC_FONT_SIZE, max_width)
-        self._wrapped_description = "\n".join(wrapped_lines)
-        self._description_height = len(wrapped_lines) * ITEM_DESC_FONT_SIZE + 10
-      return ITEM_BASE_HEIGHT + self._description_height - (ITEM_BASE_HEIGHT - ITEM_DESC_V_OFFSET) + ITEM_PADDING
-    return ITEM_BASE_HEIGHT
-
-  def get_content_width(self, total_width: int) -> int:
-    if self.action_item and self.action_item.get_width() > 0:
-      return total_width - self.action_item.get_width() - RIGHT_ITEM_PADDING
-    return total_width
-
-  def get_right_item_rect(self, item_rect: rl.Rectangle) -> rl.Rectangle:
-    if not self.action_item:
-      return rl.Rectangle(0, 0, 0, 0)
-
-    right_width = self.action_item.get_width()
-    if right_width == 0:  # Full width action (like DualButtonAction)
-      return rl.Rectangle(item_rect.x + ITEM_PADDING, item_rect.y,
-                          item_rect.width - (ITEM_PADDING * 2), ITEM_BASE_HEIGHT)
-
-    right_x = item_rect.x + item_rect.width - right_width
-    right_y = item_rect.y
-    return rl.Rectangle(right_x, right_y, right_width, ITEM_BASE_HEIGHT)
-
-
 class ListView(Widget):
   def __init__(self, items: list[ListItem]):
     super().__init__()
-    self._items = items
+    self.items = items
     self.scroll_panel = GuiScrollPanel()
-    self._font = gui_app.font(FontWeight.NORMAL)
-    self._hovered_item = -1
-    self._total_height = 0
 
   def _render(self, rect: rl.Rectangle):
-    self._update_layout_rects()
+    total_height = sum(item._get_height(int(rect.width)) for item in self.items if item.is_visible)
 
-    # Update layout and handle scrolling
-    content_rect = rl.Rectangle(rect.x, rect.y, rect.width, self._total_height)
+    # Handle scrolling
+    content_rect = rl.Rectangle(rect.x, rect.y, rect.width, total_height)
     scroll_offset = self.scroll_panel.handle_scroll(rect, content_rect)
 
-    # Handle mouse interaction
-    if self.scroll_panel.is_click_valid():
-      self._handle_mouse_interaction(rect, scroll_offset)
```

Comment:
Can you move this into generic `_handle_mouse_release`?

Reply:
Not right now. This function isn’t meant to handle events itself; it prevents other widgets from using mouse events during scrolling. Widgets using the scroll panel should call it to check if they need to handle mouse events. This is a temporary solution and will be removed once a mouse event consumption mechanism is available.